docs/reference/api/platform/Logger.md

```diff
@@ -0,0 +1,164 @@
+## Logger
+
+Mbed OS provides a set of API's that you can use to output different log level messages to STDIO at runtime. mbed_logger.h declares these functions, which are available only in debug builds.
+API's in logging module are printf-style API's which take module name and format string followed by arguments.
+
+By default all messages till LOG_LEVEL_DEBUG are enabled. If you want to enable Trace and Info level messages, then set `MBED_CONF_MAX_LOG_LEVEL` accordingly.
+
+### Debug Log Levels
+Below are various log levels supported and recommended usage.
+
+#### LOG_LEVEL_ERR_CRITICAL
+In case of critical errors when system recovery is not possible, you should use `MBED_CRIT` API. `MBED_CRIT` is equivalent to `ASSERT`, it will dump previous debug log and application will terminate. 
+
+Usage:
+```C
+uint8_t serial_tx_active(serial_t *obj) {
+    MBED_ASSERT(obj);
+    ...
+}
+Is Equivalent to 
+uint8_t serial_tx_active(serial_t *obj) {
+    if (NULL == obj) {
+        MBED_CRIT("UART", "NULL object passed %s" __FUNCTION__);
+    }
+    ...
+}
+```
+
+#### LOG_LEVEL_ERR
+You should use error level in OS, HAL, library, application, etc to report all the errors. Errors are not considered fatal by OS, library and will not terminate the application.
+
+Usage:
+```C
+Funcitons returns NULL and does not terminate the application. If error is categorized as fatal, use `MBED_CRIT` instead.
+void *operator new(std::size_t count) {
+    void *buffer = malloc(count);
+    if (NULL == buffer) {
+        MBED_ERR("OS", "Operator new out of memory");
+    }
+    return buffer;
+}
+
+Fatal error, will terminate the application if memory is not available.
+void *operator new(std::size_t count) {
+    void *buffer = malloc(count);
+    if (NULL == buffer) {
+        MBED_CRIT("OS", "Operator new out of memory");
+    }
+    return buffer;
+}
+```
+
+#### LOG_LEVEL_WARN
+You should use warning level in OS, HAL, library, application, etc to log all the warning messages.
+
+Usage:
+```C
+void dev_init() {
+    ...
+    if (DEV_ID != id) {
+        MBED_WARN ("Driver", "Incorrect Device ID");
+    }
+    ...
+}
+```
+
```

Comment:
Can you please add more examples which shows more arguments?

Reply:
https://os.mbed.com/teams/mbed_example/code/logging/file/10b2ce72c7ac/main.cpp/
We can add more samples in this example file